### ISSUE TITLE
Playground does not catch certain Shellcheck issues

### ISSUE THREAD

**Issue by stackptr** [rhysd/actionlint#364]
With `.github/workflows/test.yml`:

```yml
name: Test workflow

on:
  push:

jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - run: |
          now="$(date +%s)"
          read -r -d '' req_json <<MALFORMED_EOF
          {
            "foo": "bar"
          }
                MALFORMED_EOF
          echo "$req_json"
```

Copying and pasting into the [playground][0] yields no issues in Safari or Chrome.

Running `actionlint` (via [Nix][1]) in CLI catches an issue with malformed here document:

```
shell ❯ actionlint .github/workflows/test.yml
.github/workflows/test.yml:10:9: shellcheck reported issue in this script: SC1009:info:2:1: The mentioned syntax error was in this simple command [shellcheck]
   |
10 |       - run: |
   |         ^~~~
.github/workflows/test.yml:10:9: shellcheck reported issue in this script: SC1073:error:2:24: Couldn't parse this here document. Fix to allow more checks [shellcheck]
   |
10 |       - run: |
   |         ^~~~
.github/workflows/test.yml:10:9: shellcheck reported issue in this script: SC1039:error:6:1: Remove indentation before end token (or use <<- and indent with tabs) [shellcheck]
   |
10 |       - run: |
   |         ^~~~
.github/workflows/test.yml:10:9: shellcheck reported issue in this script: SC1072:error:9:1: Here document was not correctly terminated. Fix any mentioned problems and try again [shellcheck]
   |
10 |       - run: |
   |         ^~~~
```

As far as I can tell, there is no configuration that `actionlint` is using locally that would explain these different outputs.

[0]: https://rhysd.github.io/actionlint/
[1]: https://github.com/NixOS/nixpkgs/blob/nixos-23.05/pkgs/development/tools/analysis/actionlint/default.nix#L39

**Comment by sondrelg** [comment]
Another example, running v1.7.1 with pre-commit: https://github.com/snok/container-retention-policy/actions/runs/9641151530/job/26586217725

**Comment by rhysd** [comment]
This is a limitation of playground because shellcheck cannot run on browser. Playground runs actionlint via WASM on client side.

**Comment by sondrelg** [comment]
For my case, it was just that shellcheck is installed on github runners, and wasn't on my machine ✅

**Comment by rhysd** [comment]
Okay, so it is not related to this issue. Let me mark comments off-topic (including this).
